### PR TITLE
[ricos-theme] JSS to css-variables migration

### PR DESCRIPTION
**Pull request by Barackos**
### :rocket: New Feature
- New design implementation for the plugins toolbar - active background & hover background
- 3-Color Palette API, in addition to Wix-Palette format
  
### 🏠 Internal
- JSS deprecation, improved bundle size. replaced with CSS-Vars.

-----

3-Color Palette API:

```jsx
import { createTheme } from 'ricos-theme';
// ...
const palette = {
  bgColor: '#FFFFFF',
  textColor: '#414141',
  actionColor: '#FA6400',
};
// ...
<RicosEditor theme={createTheme({ palette })} />;
```

- Unit-Test covered
- Implemented in Storybook

### CODE REVIEW

File: packages/common/web/statics/styles/palette.scss [hunk 1]
```diff
@@ -1,15 +1,27 @@
...
 $secondary-text-color: #757575;
 
 $modal-input-color: #333333;
-$modal-warning-color: #ee5951;
-$modal-btn-confirm-primary-color: white;
-$modal-btn-confirm-secondary-color: #0261ff;
 
 $toolbar-font-color: #333333;
 $toolbar-border-color: #ededed;
+
```
Comment (Barackos) [inline packages/common/web/statics/styles/palette.scss:11]: Cannot remove all remaining `${colorName}`s, as they are still in use.
Should consider deprecating those as well in the wiring process with Nitzan.

File: packages/ricos-theme/web/src/ThemeGenerator.ts [hunk 2]
```diff
@@ -20,6 +15,24 @@ const getColorByCode = (palette: Palette, code: number): Color => {
...
+    --ricos-action-color: ${actionColor};
+    --ricos-action-color-tuple: ${toRgbTuple(actionColor)};
+    --ricos-action-color-fallback: ${adaptForeground(actionColor)};
+    --ricos-action-color-fallback-tuple: ${toRgbTuple(adaptForeground(actionColor))};
+    --ricos-action-color-fallback-bright: ${fallbackColorBright};
+    --ricos-action-color-fallback-bright-tuple: ${toRgbTuple(fallbackColorBright)};
+    --ricos-background-color: ${bgColor};
+    --ricos-background-color-tuple: ${toRgbTuple(bgColor)};
+  }\n`;
+};
```
Comment (Barackos) [inline packages/ricos-theme/web/src/ThemeGenerator.ts:34]: This is the interesting part

File: packages/ricos-theme/web/src/themes/utils.ts [hunk 3]
```diff
@@ -22,16 +22,50 @@ export function adaptForeground(actionColor: string): string {
...
+}
+
+/**
+ * Converts `hexColor` from HEX format to a CSS RGBA string
+ * @example
+ * toCssRgbA('#FFFFFF', 0.5) => 'rgba(255, 255, 255, 0.5)'
+ *
+ * @param hexColor color in HEX format
+ * @returns RGB object
+ */
```
Comment (Barackos) [inline packages/ricos-theme/web/src/themes/utils.ts:65]: Experimenting with TypeScript documentation.
@mharsat We might want to add this to other type definitions in the documentation effort, as it prompts descriptions while typing functions / fields.

Comment (mharsat) [reply packages/ricos-theme/web/src/themes/utils.ts:65]: Don't think that's specifically TS related, but that's very good to have!

File: packages/ricos-theme/web/src/ThemeGenerator.ts [hunk 4]
```diff
@@ -52,20 +65,15 @@ export default class ThemeGenerator {
 
   getStylesObject() {
```
Comment (mharsat) [inline packages/ricos-theme/web/src/ThemeGenerator.ts:66]: Looks like this doesn't return an object anymore, maybe rename?

**Review by mharsat — COMMENTED** [review]
The ricos-theme part looks great 🚀 Make sure we can confidently make all the style changes, check that what we need is covered in e2e

**Comment by oded-soffrin** [comment]
Looks good to me!
especially this part - ricos-viewer-with-theme: 227KB => 199KB

File: packages/ricos-theme/web/src/ThemeGenerator.ts (hunk 4 above)
Comment (Barackos) [reply packages/ricos-theme/web/src/ThemeGenerator.ts:66]: sounds good

File: packages/common/web/statics/styles/palette.scss [hunk 5]
```diff
@@ -1,15 +1,27 @@
-$primary-color: #607d8b;
```
Comment (lxgreen) [inline packages/common/web/statics/styles/palette.scss:1]: maybe it's too late to ask, but couldn't you just
```
$primary-color: var(--primary-color);
```
and then you don't have to replace it everywhere?

Comment (Barackos) [reply packages/common/web/statics/styles/palette.scss:1]: You're right, fixed! Never too late with `Find & Replace` 😄

**Comment by Barackos** [comment]
merged into v8 updates

**Pull request closed without merging**